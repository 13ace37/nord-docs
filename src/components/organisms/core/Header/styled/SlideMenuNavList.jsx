/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import styled from "styled-components";

import { ms } from "styles/theme";

/**
 * A flexbox list for slide menu navigation links with dynamic spacing.
 *
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 */
const SlideMenuNavList = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  font-size: ${ms(1)};

  * {
    margin: 1em;
  }
`;

export default SlideMenuNavList;
