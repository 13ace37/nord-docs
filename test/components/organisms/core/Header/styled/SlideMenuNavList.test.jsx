/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import React from "react";

import { renderWithTheme } from "nord-docs-test-utils";
import SlideMenuNavList from "organisms/core/Header/styled/SlideMenuNavList";

describe("theme styles", () => {
  test("matches the snapshot", () => {
    const { container } = renderWithTheme(<SlideMenuNavList />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
