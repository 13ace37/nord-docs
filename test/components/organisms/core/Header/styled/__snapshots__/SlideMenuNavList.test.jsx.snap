// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme styles matches the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-size: 1.125em;
}

.c0 * {
  margin: 1em;
}

<div
  class="c0"
/>
`;
